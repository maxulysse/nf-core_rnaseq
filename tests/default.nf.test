nextflow_pipeline {

    name "Test pipeline with default settings"
    script "../main.nf"

    test("Params: default") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_rnaseq_software_mqc_versions.yml")).match("software_versions") },
                { assert snapshot(
                        path("${params.outdir}/salmon/salmon.merged.transcript_counts.tsv"),
                        path("${params.outdir}/custom/out/genome_gfp.fasta"),
                        path("${params.outdir}/custom/out/genome_gfp.gtf"),
                        path("${params.outdir}/star_salmon/bigwig/RAP1_IAA_30M_REP1.forward.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/RAP1_IAA_30M_REP1.reverse.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/RAP1_UNINDUCED_REP1.forward.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/RAP1_UNINDUCED_REP1.reverse.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/RAP1_UNINDUCED_REP2.forward.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/RAP1_UNINDUCED_REP2.reverse.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/WT_REP1.forward.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/WT_REP1.reverse.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/WT_REP2.forward.bigWig"),
                        path("${params.outdir}/star_salmon/bigwig/WT_REP2.reverse.bigWig")
                    ).match("output_files")
                }
            )
        }
    }
}
