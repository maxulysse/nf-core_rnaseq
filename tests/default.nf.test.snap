{
    "output_files": {
        "content": [
            "salmon.merged.transcript_counts.tsv:md5,ff0f5be09ca7a322672c0074ba35da17",
            "genome_gfp.fasta:md5,e23e302af63736a199985a169fdac055",
            "genome_gfp.gtf:md5,c98b12c302f15731bfc36bcf297cfe28",
            "RAP1_IAA_30M_REP1.forward.bigWig:md5,0abafd7a9f9035469c003fd3dabd73e8",
            "RAP1_IAA_30M_REP1.reverse.bigWig:md5,0f1e9ac71fc0b99785f06ecb860ef00e",
            "RAP1_UNINDUCED_REP1.forward.bigWig:md5,09e8d65e21ac92e9a3b78afe3acdf28b",
            "RAP1_UNINDUCED_REP1.reverse.bigWig:md5,8b002d23911f95f1c5b9965c2e7d0737",
            "RAP1_UNINDUCED_REP2.forward.bigWig:md5,eede9cac23d47c4685dc13507e97bfb9",
            "RAP1_UNINDUCED_REP2.reverse.bigWig:md5,a835e0ac51231016c52de94c18ebd466",
            "WT_REP1.forward.bigWig:md5,92ad01ddd1dbb7f6d7e2eae23df58c7c",
            "WT_REP1.reverse.bigWig:md5,dcac37ef5fff108afd0632388919dcf7",
            "WT_REP2.forward.bigWig:md5,d10cbea27b82745b46077191d3ff2820",
            "WT_REP2.reverse.bigWig:md5,2b3a15b511efdbc838574200ee2bec0c"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-27T15:38:12.586958"
    },
    "software_versions": {
        "content": [
            "{BBMAP_BBSPLIT={bbmap=39.01}, BEDTOOLS_GENOMECOV_FW={bedtools=2.31.1}, CAT_FASTQ={cat=8.3}, CUSTOM_CATADDITIONALFASTA={python=3.9.5}, CUSTOM_GETCHROMSIZES={getchromsizes=1.2}, CUSTOM_TX2GENE={python=3.9.5}, DESEQ2_QC_PSEUDO={r-base=4.0.3, bioconductor-deseq2=1.28.0}, DESEQ2_QC_STAR_SALMON={r-base=4.0.3, bioconductor-deseq2=1.28.0}, DUPRADAR={bioconductor-dupradar=1.32.0}, FASTQC={fastqc=0.12.1}, FQ_SUBSAMPLE={fq=0.9.1 (2022-02-22)}, GTF2BED={perl=5.26.2}, GTF_FILTER={python=3.9.5}, GUNZIP_ADDITIONAL_FASTA={gunzip=1.1}, GUNZIP_GTF={gunzip=1.1}, MULTIQC_CUSTOM_BIOTYPE={python=3.9.5}, PICARD_MARKDUPLICATES={picard=3.1.1}, QUALIMAP_RNASEQ={qualimap=2.3}, RSEQC_BAMSTAT={rseqc=5.0.2}, RSEQC_INFEREXPERIMENT={rseqc=5.0.2}, RSEQC_INNERDISTANCE={rseqc=5.0.2}, RSEQC_JUNCTIONANNOTATION={rseqc=5.0.2}, RSEQC_JUNCTIONSATURATION={rseqc=5.0.2}, RSEQC_READDISTRIBUTION={rseqc=5.0.2}, RSEQC_READDUPLICATION={rseqc=5.0.2}, SALMON_QUANT={salmon=1.10.1}, SAMTOOLS_FLAGSTAT={samtools=1.2}, SAMTOOLS_IDXSTATS={samtools=1.2}, SAMTOOLS_INDEX={samtools=1.2}, SAMTOOLS_SORT={samtools=1.2}, SAMTOOLS_STATS={samtools=1.2}, SE_GENE={bioconductor-summarizedexperiment=1.32.0}, STAR_ALIGN={star=2.7.10a, samtools=1.18, gawk=5.1.0}, STAR_GENOMEGENERATE={star=2.7.10a, samtools=1.18, gawk=5.1.0}, STRINGTIE_STRINGTIE={stringtie=2.2.1}, SUBREAD_FEATURECOUNTS={subread=2.0.1}, TRIMGALORE={trimgalore=0.6.7, cutadapt=3.4}, TXIMETA_TXIMPORT={bioconductor-tximeta=1.20.1}, UCSC_BEDCLIP={ucsc=377}, UCSC_BEDGRAPHTOBIGWIG={ucsc=445}, UNTAR_SALMON_INDEX={untar=1.3}, Workflow={nf-core/rnaseq=v3.15.0dev}}"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-27T15:38:12.576022"
    }
}